pandas/io/parsers.py

```diff
@@ -855,14 +855,14 @@ def _get_options_with_defaults(self, engine):
 
     def _check_file_or_buffer(self, f, engine):
         # see gh-16530
-        if is_file_like(f):
+        if is_file_like(f) and engine != "c" and not hasattr(f, "__next__"):
```

Comment:
what is file_like but doesn't have a ``__next__``?

Reply:
I think `mmap` is such a case (that's why there is a `mmap` wrapper in io/common).

Reply:
this is just flattening two nested if statements